dima/react-movies/src/App.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { Component } from "react";
+import MainPage from "./pages/mainpage";
+import MoviePage from "./pages/moviepage";
+import MovieInterface from "./interfaces/movieInterface";
+import SortProperty from "./enums/SortProperty";
+import FilterProperty from "./enums/FilterPropery";
+import PageNotFound from "./pages/pageNotFound";
+
+import {
+  BrowserRouter as Router,
+  Redirect,
+  Route,
+  Switch,
+} from "react-router-dom";
+import "./App.css";
+
+interface AppState {
+  movie: MovieInterface | null;
+  movies: MovieInterface[];
+  isMovieExisted: boolean;
+  isMoviesExisted: boolean;
+  currentSortType: string;
+}
+
+class App extends Component<{}, AppState> {
+  state: AppState = {
+    movie: null,
+    movies: [],
+    isMovieExisted: true,
+    isMoviesExisted: true,
+    currentSortType: SortProperty.date,
+  };
+
+  setCurrentSortType = (currentSortType: string): void => {
+    this.setState({ currentSortType });
+  };
+  searchMovies = (value: string, category: string): void => {
+    const { movies } = this.state;
+    if (!value) {
+      return;
+    }
+    if (category === FilterProperty.title) {
+      this.setState({
+        movies: movies.filter((movie) =>
+          movie.title.toLowerCase().includes(value.toLowerCase())
+        ),
+      });
+    } else {
+      this.setState({
+        movies: movies.filter((movie) =>
+          movie.genres
+            .map((genre) => genre.toLocaleLowerCase())
+            .includes(value.toLowerCase())
+        ),
+      });
+    }
+  };
+  setIsMoviesExisted = (isMoviesExisted: boolean): void => {
+    this.setState({ isMoviesExisted });
+  };
+  setIsMovieExisted = (isMovieExisted: boolean): void => {
+    this.setState({ isMovieExisted });
+  };
+  setMovies = (movies: MovieInterface[]): void => {
+    this.setState({ movies });
+  };
+  setMovie = (movie: MovieInterface): void => {
+    this.setState({ movie });
+  };
+  // sortMovies(type: string):void {
+  //   const { movies } = this.state;
+  //   if (type === SortProperty.date) {
+  //     this.setState({
+  //       movies: [...movies].sort((a, b) => {
+  //         return (
+  //           Number.parseInt(b.release_date) - Number.parseInt(a.release_date)
+  //         );
+  //       }),
+  //     });
+  //   } else {
+  //     this.setState({
+  //       movies: [
+  //         ...movies.sort((a, b) => {
+  //           return b.vote_average - a.vote_average;
+  //         }),
+  //       ],
+  //     });
+  //   }
+  // }
+
+  render() {
+    const {
+      movies,
+      movie,
+      currentSortType,
+      isMovieExisted,
+      isMoviesExisted,
+    } = this.state;
+    return (
+      <>
+        <Router>
+          <Switch>
+            <Route
+              exact
+              path={["/", "/search"]}
+              render={(props) => (
+                <MainPage
+                  route={{ ...props }}
```

Comment:
route props better not to assign to some property. Because route props are well known, and I expect to have **this.props.location** or **this.props.match** instead of this.props.route... 
Do {...props} instead.